python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -90,6 +97,15 @@ def execute_job_backfill_iteration(
             yield None
             time.sleep(CHECKPOINT_INTERVAL)
         else:
+            backfill_runs = instance.get_runs(
+                RunsFilter(tags=DagsterRun.tags_for_backfill_id(backfill.backfill_id))
+            )
```

Comment:
suggest adding this as a filter to the get_runs call (like asset backfills do: https://github.com/dagster-io/dagster/blob/286a583bcae8d3170820896d2b007221b7b43a50/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L1111-L1115) rather than fetching all runs and then filtering in memory

Reply:
should the QUEUED status also get included in the filter? if we've got a couple runs queued and every other run is finished i think we'd still want the backfill to be in progress right?

Reply:
.... yes. I think you may have just identified a bug in the asset backfill logic I posted.

You can also add a limit=1 since we only care if any of them exist

Reply:
ah Ok, it's not a bug in the asset backfill case because that also checks if there are 'no runs to cancel' which would only be hit if there are no longer any queued runs.

Reply:
cool cool - the filtering is updated for this PR then - and the limit thing